src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsallocations/client/AssessRisksNeedsApiClient.kt

```diff
@@ -1,49 +1,39 @@
 package uk.gov.justice.digital.hmpps.hmppsallocations.client
 
+import kotlinx.coroutines.flow.Flow
+import org.springframework.http.HttpHeaders
 import org.springframework.http.HttpStatus
 import org.springframework.web.reactive.function.client.WebClient
-import reactor.core.publisher.Flux
+import org.springframework.web.reactive.function.client.awaitBodyOrNull
+import org.springframework.web.reactive.function.client.bodyToFlow
 import reactor.core.publisher.Mono
 import uk.gov.justice.digital.hmpps.hmppsallocations.domain.RiskPredictor
 import uk.gov.justice.digital.hmpps.hmppsallocations.domain.RoshSummary
 
 class AssessRisksNeedsApiClient(private val webClient: WebClient) {
 
-  fun getRosh(crn: String): Mono<RoshSummary> {
+  suspend fun getRosh(crn: String, authToken: String): RoshSummary? {
```

Comment:
couldn't figure out how use filter on the actual webclient when initialising so had to pass through from controller